modules/user/user.service.js

```diff
@@ -0,0 +1,155 @@
+const { AuthenticationError, UserInputError } = require('apollo-server');
+const bcrypt = require('bcryptjs');
+const User = require('./user.model');
+const {
+  validateRegisterInput,
+  validateLoginInput,
+} = require('../../utils/validateUser');
+const generateToken = require('../../utils/createToken');
+
+class UserService {
+  getAllUsers() {
+    return User.find();
+  }
+
+  getUserById(id, auth) {
+    if (auth.errors) throw new Error(auth.errors);
+    return User.findById(id);
```

Comment:
Please, add check that if user doesn't exist then throw a `NotFound` exception

Reply:
fixed